--- collectors/powerdns/powerdns.py.orig	2017-09-05 14:43:49.511625749 -0400
+++ collectors/powerdns/powerdns.py	2017-09-05 14:49:23.059110680 -0400
@@ -53,7 +53,7 @@
             self.log.error("%s is not executable", self.config['bin'])
             return False
 
-        command = [self.config['bin'], 'list']
+        command = ['timeout', '-s', '9', '5', self.config['bin'], 'list']
 
         if str_to_bool(self.config['use_sudo']):
             command.insert(0, self.config['sudo_cmd'])
